xray/backends/netCDF4_.py

```diff
@@ -164,6 +171,18 @@ def set_variable(self, name, variable):
             fill_value = None
 
         encoding = variable.encoding
+        endian = encoding.get('endian', _endian_lookup[datatype.byteorder])
+        data = variable.values
+        if endian is not 'native':
```

Comment:
do we really need to convert all non-native types? the comment below says only big endian is problematic.